### PR TITLE
macros with editing errors should not log tracebacks; fixes #1687

### LINKED ISSUE #1687: macros with editing errors should not fill the log with tracebacks
Remove code similar to:

    raise ValueError("Bad timestamp %r: %s" % (args, err))

that fills the log with  entries similar to:

    Traceback (most recent call last):
      File "C:\GIT\moin\src\moin\macros\Date.py", line 69, in macro
        tm = self.parse_time(tm)
      File "C:\GIT\moin\src\moin\macros\Date.py", line 53, in parse_time
        tm = float(args)
    ValueError: could not convert string to float: 'qqq'

and replace it with code that creates an error message so the editor can fix the problem.

### PR DESCRIPTION
**Pull request by RogerHaase**

### CODE REVIEW

File: src/moin/macros/_base.py [hunk 1]
```diff
@@ -85,13 +85,24 @@ def extract_h1(item_name):
...
+    :param alternative: full text of macro as passed in MacroBase __call__
+    :param Severity: attention, caution, danger, error, hint, important, note, or tip
+    :returns: formatted text of macro, error message
     """
-    return html.div(attrib={html.class_: f"{severity} moin-nowiki"}, children=msg)
+    if severity not in "attention caution danger error hint important note tip".split():
+        severity = "error"
+        msg = (
+            _("Invalid severity, must be one of: ") + "attention, caution, danger, error, hint, important, note, or tip"
+        )
```
Comment (ThomasWaldmann) [inline src/moin/macros/_base.py:101]: does severity always come from the code or also from wiki content? if it comes from the code, guess just crashing with a ValueError would be more appropriate here (the wiki user can't fix errors in the code).

**Review by ThomasWaldmann — COMMENTED** [review]
some stuff i found.

File: src/moin/macros/Date.py [hunk 2]
```diff
@@ -32,41 +33,42 @@ def parse_time(self, args):
...
-            tm = self.parse_time(stamp)
-        return show_time.format_date(tm)
+            tm = arguments[0]
+        try:
+            if tm:
+                tm = self.parse_time(tm)
+            return show_time.format_date(tm)
+        except (AttributeError, OSError, AssertionError, ValueError, OverflowError):
+            err_msg = _("Invalid input parameter: null, float, int, or ISO 8601 formats are accepted.")
+            return fail_message(err_msg, alternative)
```
[the remark this answers is not in the document]
Comment (RogerHaase) [reply src/moin/macros/Date.py:74]: Normally I would agree, but in this case errors similar to

<<Date(1434563755888)>>

pass parse_time but fail in show_time. So easiest way was to wrap both in the try.

File: src/moin/macros/Date.py [hunk 3]
```diff
@@ -32,41 +33,42 @@ def parse_time(self, args):
...
-            tm = None
+            tm = None  # show today's date
         else:
-            stamp = arguments[0]
-            tm = self.parse_time(stamp)
-        return show_time.format_date(tm)
+            tm = arguments[0]
+        try:
+            if tm:
+                tm = self.parse_time(tm)
```
[the remark this answers is not in the document]
Comment (RogerHaase) [reply src/moin/macros/Date.py:70]: If tm is None, there is nothing to parse. show_time will handle None and integers

File: src/moin/macros/Date.py [hunk 4]
```diff
@@ -19,10 +20,10 @@ def parse_time(self, args):
         Parse a time specification argument for usage by Date and DateTime macro.
         Not all ISO 8601 format variations are accepted as input.
 
-        :param args: float/int UNIX timestamp or ISO 8601 formatted date time:
+        :param args: float/int UNIX timestamp or null or ISO 8601 formatted date time:
```
[the remark this answers is not in the document]
Comment (RogerHaase) [reply src/moin/macros/Date.py:23]: agree, will fix here and elsewhere

File: src/moin/macros/FontAwesome.py [hunk 5]
```diff
@@ -20,14 +20,17 @@
...
 
 class Macro(MacroInlineBase):
     def macro(self, content, arguments, page_url, alternative):
         args = arguments[0] if arguments else ""
         if not args:
-            raise ValueError("Missing font name")
+            err_msg = _("Missing font name, syntax is <<FontAwesome(name,color,size)>>")
+            return fail_message(err_msg, alternative)
+
         args = args.split(",")
```
[the remark this answers is not in the document]
Comment (RogerHaase) [reply src/moin/macros/FontAwesome.py:34]: fixed

File: src/moin/macros/_base.py (hunk 1 above)
Comment (RogerHaase) [reply src/moin/macros/_base.py:101]: Only from code, fixed.  Thank you.

**Pull request merged**